### ISSUE TITLE
buildRunner is deprecated

### ISSUE THREAD

**Issue by Kernald** [buildfoundation/bazel_rules_detekt#103]
Building rules_detekt 0.4.0 raises a warning:

```
external/rules_detekt/detekt/wrapper/src/main/kotlin/io/buildfoundation/bazel/detekt/execute/Detekt.kt:13:13: warning: 'buildRunner(Array<String>, PrintStream, PrintStream): Executable' is deprecated. Don't build a runner yourself.
            buildRunner(args, outputPrinter, errorPrinter).execute()
            ^
```

What are the alternatives here?

**Comment by arturdryomov** [comment]
There is a new Tooling API briefly introduced in #90. However, it broke error propagation which wasn’t detected due to an instrumentation tests bug. This was resolved via rolling back to the CLI API in #102. I’ll have to look around how the Tooling API should be used. It’s a bit confusing since the Gradle plugin [seem to use the CLI API](https://github.com/detekt/detekt/blob/56b5f7726c831f181d6c2b8a4a02e2b0389df8bf/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt#L51-L56). Also it is [undocumented by design](https://github.com/detekt/detekt/issues/2860#issuecomment-711210765) at this point.
